api-client/spec/api/extensions/get_into_teaching_api_client/api_client_spec.rb

```diff
@@ -0,0 +1,145 @@
+RSpec.describe Extensions::GetIntoTeachingApiClient::ApiClient do
+  let(:host) { "host.api" }
+  let(:endpoint) { "endpoint" }
+  let(:get_endpoint) { "https://#{host}/#{endpoint}/api/pick_list_items/candidate/channels" }
+  let(:post_endpoint) { "https://#{host}/#{endpoint}/api/candidates/access_tokens" }
+  let(:token) { "test" }
+  let(:data) { [{ id: 123, value: "test" }] }
+  let(:cache_store) { ActiveSupport::Cache.lookup_store(:null_store) }
+
+  def perform_get_request
+    GetIntoTeachingApiClient::PickListItemsApi.new.get_candidate_channels
+  end
+
+  def perform_post_request
+    existing_candidate = GetIntoTeachingApiClient::ExistingCandidateRequest.new(email: "test@test.com")
+    GetIntoTeachingApiClient::CandidatesApi.new.create_candidate_access_token(existing_candidate)
+  end
+
+  before do
+    GetIntoTeachingApiClient.configure do |config|
+      config.host = host
+      config.base_path = endpoint
+      config.api_key["Authorization"] = token
+      config.cache_store = cache_store
+    end
+    cache_store&.clear
+  end
+
+  describe "formatting DateTime/Time attributes in query string parameters" do
```

Comment:
This and the `describe` block on line 59 are the only additions/changes to this file.